### DISCUSSION TITLE
What is the default optimization level in general people use to compile for OpenRadioss?

### DISCUSSION THREAD

**Discussion by rashinde45** [OpenRadioss/OpenRadioss#3087] [Users]
I know this is very generic question, but what is the recommended optimization level to compile OpenRadioss. Are upstream people used O2 or they use O3 and Ofast also to compile the OpenRadioss. is the OpenRadioss app have vectorizable patterns? The reason I am asking if I run with O2 rather than O3 or Ofast, will I miss any auto vectorization ?

**Comment by laurent-altr** [comment]
We focus on using vectorizable patterns when developing OpenRadioss.

For the engine binary, using Intel's compilers, we use -axSSE3,COMMON-AVX512 -no-fma -O3 -fp-model precise -fimf-use-svml=true. The -no-fma and -fp-model precise flags slightly reduce performance but ensure numerical reproducibility and accuracy.

If you're building and running OpenRadioss on the same machine and prioritize speed over numerical accuracy, you can try -xHost -Ofast -fimf-use-svml=true. Keep in mind that it sacrifices numerical reproducibility for performance.
